### ISSUE TITLE
Zod's .transform not affecting .getValues

### ISSUE THREAD

**Issue by andresespinosapc** [react-hook-form/resolvers#568]
**Describe the bug**
When I use `.transform` in a Zod schema, the type of the output is the output of the transform, but the input is not actually transformed.

**To Reproduce**
1. Create a schema that uses a transform on a field
2. Compare the values of that field using the data got on `handleSubmit` with the one from `getValues()`

**Codesandbox link (Required)**
https://codesandbox.io/s/react-hook-form-zod-resolver-example-forked-x9lqe1?file=/src/App.js

**Expected behavior**
The values should be the same. The values from `getValues()` should be transformed.

**Desktop (please complete the following information):**

- OS: macOS 13.3.1
- Browser Chrome
- Version 2.9.10

**Comment by kasparkallas** [comment]
FYI, the transformed value is passed to `onValid` though for the `handleSubmit`.

**Comment by marcebdev** [comment]
Yup the real issue is the typings, using `.transform` will either break the types for things that access the form's state before submit(for instance the example above) or it will break the submit type.

**Comment by jorisre** [comment]
This represents the standard behavior as elucidated [in the documentation](https://react-hook-form.com/docs/useform/getvalues):

> An optimized helper for reading form values. The difference between watch and getValues is that getValues will not trigger re-renders or subscribe to input changes.

You may consider utilizing the `watch` utility to observe modifications.

Updated CodeSandbox
https://codesandbox.io/s/react-hook-form-zod-resolver-example-forked-3ppwl2?file=/src/App.js

**Comment by ctjhoa** [comment]
@jorisre Excuse me but the point is that `watch` and `getValues` should return a `File` not a `FileList`.
In your latest codesandbox you demonstrate exactly the issue that `watch` return a non-transformed value.

**Comment by jorisre** [comment]
I appreciate the clarification. The problem doesn't lie within the resolver; it correctly returns the transformed value. The concern lies with react-hook-form. The `console.log` confirms that the value undergoes transformation as intended by the resolver.

https://codesandbox.io/s/react-hook-form-zod-resolver-example-forked-3ppwl2?file=/src/App.js
